src/vdaf/poplar1.rs

```diff
@@ -1245,6 +1246,52 @@ impl<P: Xof<SEED_SIZE>, const SEED_SIZE: usize> Aggregator<SEED_SIZE, 16>
             output_shares,
         )
     }
+
+    /// Validates that no aggregation parameter with the same level as `cur` has been used with the
+    /// same input share before. `prev` contains the aggregation parameters used for the same input.
+    ///
+    /// # Panics
+    /// Panics if `prev.len() > 1`
+    fn is_agg_param_valid(cur: &Poplar1AggregationParam, prev: &[Poplar1AggregationParam]) -> bool {
+        assert!(
+            prev.len() <= 1,
+            "list of previous aggregation parameters must be of size 0 or 1"
+        );
+
```

Comment:
For heavy hitters applications using Poplar1, we'd want the full list of previous aggregation parameters to be passed in here. Thus, we shouldn't have a cap on the length of this list.

While Poplar1 only cares about the most recent previous aggregation parameter, other VDAFs may care about more, so the contract should be that they all get passed in. In particular, Mastic will look at all of them, and check that exactly one aggregation parameter has the `do_weight_check` flag set.
```suggestion
    fn is_agg_param_valid(cur: &Poplar1AggregationParam, prev: &[Poplar1AggregationParam]) -> bool {
```

Reply:
Agreed this should be general. This panic condition and check is only for Poplar1, though. Isn't that the appropriate amount of specialization?

Reply:
I don't think this is correct for Poplar1. Poplar1 expects to be run many times on the same reports, not just twice, in order to carry out the heavy hitters algorithm.

Reply:
I see, ok. Why does the validity check only look at the last elem then?

Reply:
Oh, I think I understand. There's an implicit requirement that the agg params are all sorted in least to most recently used, and that `is_valid()` was called at every new invocation. I've written that assumption in the comments